### PR TITLE
Update RSpec configuration for testing fun times

### PR DESCRIPTION
**Pull request by orien**
* require spec_helper via the .rspec file
* format via instafail unless running just one spec file
* format via documentation otherwise
* don't let RSpec monkey patch the world
* don't allow stubbing methods that don't exist on partial mocks
* print out the 5 slowest tests

### CODE REVIEW

File: spec/active_record/locking_extensions_spec.rb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # encoding: utf-8
-require 'spec_helper'
```
Comment (asellitt) [inline spec/active_record/locking_extensions_spec.rb]: does spec_helper get automagically included?

**Comment by asellitt** [comment]
+1

disk io errors only for sqlite...seems like travis failed here...

File: spec/active_record/locking_extensions_spec.rb (hunk 1 above)
Comment (orien) [inline spec/active_record/locking_extensions_spec.rb]: Not one for reading the PR description are you :wink:
It's specified in the `.rspec` file.

Comment (asellitt) [inline spec/active_record/locking_extensions_spec.rb]: read the README!? what sorcery is this?!

**Pull request merged**